src/ParseNatSpec.js

```diff
@@ -0,0 +1,75 @@
+
+/**
+ * Parse a full text comment to an object that can be easly consumed
+ * @param comment full text comment
+ */
+module.exports = function (comment) {
+  let isMultiline = false;
+
+  // remove comments definers
+  // the comments can be multiline of single line
+  if (comment.substring(0, 3) === '/**') {
+    isMultiline = true;
+    // if it is a multiline, remove line breaks, "/**"" (the comment begin)
+    // "*/"" (the comment end), and "* @" at the begining of each comment
+    comment = comment.replace(/(\r|\n|\/\*\*|\*\/|\*(?=[ ]?@))/g, '');
+  } else if (comment.substring(0, 3) === '///') {
+    // if it is a single line,
+    // remove just line breakers and "///" (the comment begin)
+    comment = comment.replace(/(\r|\n|\/\/\/)/g, '');
+  }
+
+  // split text by comment type
+  const splitComments = comment
+    .split(/@(title|author|notice|dev|param|return)/g);
+  // let's start a map
+  const resultComments = new Map();
+
+  // iterate through all the split comments
+  // start in the second result (because the first one is usually nothing)
+  // then iterate each two elements, because since the split is done using
+  // the natspec type
+  for (let x = 1; x < splitComments.length; x += 2) {
+
+    // verify if the type exists in the map
+    let previousValue = resultComments.get(splitComments[x]);
```

Comment:
For clarity, please extract `splitComments[x]` to a variable named `key` or similar, and extract `splitComments[x + 1]` to `value` or similar.

Reply:
great, done